api/src/main/java/org/pmiops/workbench/db/dao/ParticipantDao.java

```diff
@@ -0,0 +1,16 @@
+package org.pmiops.workbench.db.dao;
+
+import org.pmiops.workbench.db.model.Participant;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Page;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+import java.util.List;
+
+public interface ParticipantDao extends CrudRepository<Participant, Long> {
+
+    Page<Participant> findParticipantByParticipantKey_CohortIdAndParticipantKey_CdrVersionId(@Param("cohortId") long cohortId,
```

Comment:
So this works well enough if we actually materialize cohorts fully in the table you added when reviewing them. However that means inserting a bunch of rows into Cloud SQL at the start of review; possibly a million if the cohort is large. I'm not sure that's what we want. Adding @jmandel for his thoughts.

Another option is to start with nothing in the database for cohortId + cdrVersionId, and merge the participant list from the materialized cohort in BigQuery with status values pulled from this table. (If they order by status, this would entail reading by status here and then querying BigQuery for the rows.)

Reply:
This is definitely a performance optimization question -- namely, is it "too expensive" to create `unreviewed` rows for each participant in each CDR version for each cohort that a user ever defines. Given that some researchers will never really conduct a review, this could be a lot of rows without any value.

My intuition is that we might do well to only write data when a participant is actually reviewed -- but from the UX perspective, this decision wouldn't necessarily have an impact.

Reply:
As discussed offline: @jmandel and I are OK with this approach. For now we should add an endpoint for starting a cohort review, that reads from BQ and inserts participant IDs into the table; if the cohort size is larger than some configurable value (e.g. 10K) we can just fail that request.

Eventually we should probably have some way of viewing a sample of participants in the larger cohorts, even if you can't manually indicate who is in or out of them. But we don't have to worry about that now.

Reply:
@jmandel, I don't think we actually would write out rows until the researcher said they were starting a review in the UI. If no review has been started, we just query BQ and don't filter on status anywhere.

We also discussed a use case for manually picking people out of a large cohort -- if you want to use the cohort for a starting set of people, and then manually review to pick out the people you actually want to include (with no intention of using anybody you haven't manually selected for inclusion.)

So we probably need to support the "review random 10K participants", and just make it clear in our documentation that you're not necessarily getting everyone that way.

Reply:
Agreed.